packages/core/src/types/context.ts

```diff
@@ -140,13 +140,33 @@ export type KeystoneDbAPI<KeystoneListsTypeInfo extends Record<string, BaseListT
 
 export type KeystoneGraphQLAPI = {
   schema: GraphQLSchema;
-  run: (args: GraphQLExecutionArguments) => Promise<Record<string, any>>;
-  raw: (args: GraphQLExecutionArguments) => Promise<ExecutionResult<Record<string, any>>>;
+  run: <TData = any, TVariables = Record<string, any>>(
+    args: GraphQLExecutionArguments<TData, TVariables>
+  ) => Promise<ExecutionResult<TData, TVariables>['data']>;
+  raw: <TData = any, TVariables = Record<string, any>>(
+    args: GraphQLExecutionArguments<TData, TVariables>
+  ) => Promise<ExecutionResult<TData, TVariables>>;
 };
 
-type GraphQLExecutionArguments = {
-  query: string | DocumentNode;
-  variables?: Record<string, any>;
+export interface TypedDocumentNode<
```

Comment:
We should import this from `@graphql-typed-document-node/core`, not duplicate it. (the only dependency it has is a peer dep on `graphql` so I'm not worried about dependency bloat from it)

Reply:
Cool, I was wondering about that.. I'll make that change now...